lib/floe/workflow/runner/kubernetes.rb

```diff
@@ -102,6 +102,64 @@ def cleanup(runner_context)
           delete_secret(secret) if secret
         end
 
+        def wait(timeout: nil, events: %i[create update delete])
+          retry_connection = true
+
+          begin
+            watcher = kubeclient.watch_pods(:namespace => namespace)
+
+            retry_connection = true
+
+            if timeout.to_i > 0
+              timeout_thread = Thread.new do
+                sleep(timeout)
+                watcher.finish
+              end
+            end
+
+            watcher.each do |notice|
+              if notice.type == "ERROR"
+                message = notice.object&.message
+                code    = notice.object&.code
+                reason  = notice.object&.reason
+
+                logger.warn("Received [#{code} #{reason}], [#{message}]")
+                break
+              end
+
+              event = kube_notice_type_to_event(notice.type)
+              next unless events.include?(event)
+
+              pod = notice.object
+              container_ref   = pod.metadata.name
+              container_state = pod.to_h[:status].deep_stringify_keys
```

Comment:
something like the parse_notice code would be nice here.
Think that would make CodeClimate much happier.

This is probably good enough for now and can be done later